### PR TITLE
create conversation table

### PR DESCRIPTION
**Pull request by tracytrle**

### CODE REVIEW

File: medicai/conversation/routes.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from flask_cors import CORS, cross_origin
+from medicai.models.conversation import Conversation
+from medicai.models.user import User
+from medicai.extensions import db
+from medicai.conversation import bp
+from datetime import datetime
+
+@bp.route('/getLatest/<int:userId>', methods=['GET'])
+@cross_origin()
+def get_latestConversation(userId):
```
Comment (travistrle) [inline medicai/conversation/routes.py]: `get_latest_conversation`

File: medicai/conversation/routes.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    current_user = User.query.filter_by(id=userId).first()
```
Comment (travistrle) [inline medicai/conversation/routes.py]: `user = User.query.get(id)` is better than doing filter and get first.

File: medicai/conversation/routes.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    if current_user is None:
+        return jsonify(message="User not found"), 400
+    # query Conversation table to get the latest conversation Id
+    latest_conversation = Conversation.query.filter_by(userId=userId).order_by(Conversation.created_at.desc()).first()
```
Comment (travistrle) [inline medicai/conversation/routes.py:18]: remember to index userId if we query this column oftenly

File: medicai/conversation/routes.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    if latest_conversation is None or latest_conversation.sender: 
+        # create a new conversation and return it
+        new_conversation = Conversation(
+            userId=userId,
+            conversationId=(latest_conversation.conversationId + 1) if latest_conversation else 1,
```
Comment (travistrle) [inline medicai/conversation/routes.py:24]: I prefer using UUID and not using Integer.

File: medicai/conversation/routes.py (hunk 2 above)
Comment (tracytrle) [reply medicai/conversation/routes.py]: fixed

File: medicai/conversation/routes.py (hunk 1 above)
Comment (tracytrle) [reply medicai/conversation/routes.py]: fixed

**Review by travistrle — APPROVED** [review]

**Pull request merged**